### PR TITLE
ENG-8369: Fix policy docs to inform how to use 'any' for number of rows

### PR DESCRIPTION
**Pull request by zero101010**
## Description of the change

> Update documentation to explain how the user can set `any` as values in rows, and change the default example to set one of those values as `-1`

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing

> I did a policy rule with the `rows` as `-1` and check in Jeeves if the value was `any`

### CODE REVIEW

File: docs/resources/policy_rule.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ resource "cyral_policy_rule" "some_resource_name" {
 - `deletes` - (Optional) A contexted rule for accesses of the type `delete`.
 - `hosts` - (Optional) Hosts specification that limits access to only those users connecting from a certain network location.
 - `rate_limit` - (Optional) Rate Limit specifies the limit of calls that a user can make within a given time period.
+- `rows` - (Optional) Rows specification how many rows can be return to the policy rules, we can use integer numbers to define rows, but if you expect return `any` number of rows is only necessary set rows as `-1`.
```
Comment (wcmjunior) [inline docs/resources/policy_rule.md:68]: Both `rate_limit` and `rows` should be moved to a proper section as these are attributes that are part of inner blocks, not the root of the resource. Please see how this is done [here](https://github.com/cyralinc/terraform-provider-cyral/blob/main/docs/resources/repository_local_account.md). Also, also please add the description to the `Description` in the schema definition itself. See [this reference](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource_cyral_policy_rule.go#L101).

Comment (zero101010) [reply docs/resources/policy_rule.md:68]: Perfect, I will update with this changes

File: cyral/resource_cyral_policy_rule.go [hunk 2]
```diff
@@ -89,8 +89,9 @@ func resourcePolicyRule() *schema.Resource {
 					},
 				},
 				"rows": {
-					Type:     schema.TypeInt,
-					Required: true,
+					Description: "Rows specification how many rows can be return to the policy rules, we can use integer numbers to define rows, but if you expect return `any` number of rows is only necessary set rows as `-1`",
```
Comment (wcmjunior) [inline cyral/resource_cyral_policy_rule.go:92]:
```suggestion
					Description: "How many rows can be return by the policy rule. Use positive integer numbers to define how many rows. If you want to define `any` number of rows, set as `-1`.",
```

File: docs/resources/policy_rule.md [hunk 3]
```diff
@@ -76,6 +75,21 @@ resource "cyral_policy_rule" "some_resource_name" {
 
 For more information, see the [Policy Guide](https://cyral.com/docs/policy#the-rules-block-of-a-policy).
 
+### Reads
+
+- `rate_limit` - (Optional) Rate Limit specifies the limit of calls that a user can make within a given time period.
+- `rows` - (Required) Rows specification how many rows can be return to the policy rules, we can use integer numbers to define rows, but if you expect return `any` number of rows is only necessary set rows as `-1`.
```
Comment (wcmjunior) [inline docs/resources/policy_rule.md:81]:
```suggestion
- `rows` - (Required) How many rows can be return by the policy rule. Use positive integer numbers to define how many rows. If you want to define `any` number of rows, set as `-1`.
```

File: docs/resources/policy_rule.md [hunk 4]
```diff
@@ -76,6 +75,21 @@ resource "cyral_policy_rule" "some_resource_name" {
...
+
+### Updates
+
+- `rate_limit` - (Optional) Rate Limit specifies the limit of calls that a user can make within a given time period.
+- `rows` - (Required) Rows specification how many rows can be return to the policy rules, we can use integer numbers to define rows, but if you expect return `any` number of rows is only necessary set rows as `-1`.
```
Comment (wcmjunior) [inline docs/resources/policy_rule.md:86]:
```suggestion
- `rows` - (Required) How many rows can be return by the policy rule. Use positive integer numbers to define how many rows. If you want to define `any` number of rows, set as `-1`.
```

File: docs/resources/policy_rule.md [hunk 5]
```diff
@@ -76,6 +75,21 @@ resource "cyral_policy_rule" "some_resource_name" {
...
+
+### Deletes
+
+- `rate_limit` - (Optional) Rate Limit specifies the limit of calls that a user can make within a given time period.
+- `rows` - (Required) Rows specification how many rows can be return to the policy rules, we can use integer numbers to define rows, but if you expect return `any` number of rows is only necessary set rows as `-1`.
```
Comment (wcmjunior) [inline docs/resources/policy_rule.md:91]:
```suggestion
- `rows` - (Required) How many rows can be return by the policy rule. Use positive integer numbers to define how many rows. If you want to define `any` number of rows, set as `-1`.
```

**Review by wcmjunior — COMMENTED** [review]
See suggestions on how to make `rows` descriptions simpler.

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**